### PR TITLE
[Feature] v2 for Countries and All endpoints

### PR DESCRIPTION
**Pull request by buster95**
all functionality for v1 endpoint is available this update is for endpoints navigation
improvements in route navigation between actual and yesterday data

**`ALL endpoint`**
```javascript
/v2/all                 [by default show actual data]
/v2/all?yesterday       [show yesterday data]
/v2/all?yesterday=1     [show yesterday data]
/v2/all?yesterday=0     [show actual data]
/v2/all?yesterday=true  [show yesterday data]
/v2/all?yesterday=false [show actual data]
```
**`COUNTRIES endpoint`**
```javascript
/v2/countries                 [by default show actual data]
/v2/countries?yesterday       [show yesterday data]
/v2/countries?yesterday=1     [show yesterday data]
/v2/countries?yesterday=0     [show actual data]
/v2/countries?yesterday=true  [show yesterday data]
/v2/countries?yesterday=false [show actual data]
```
**`SPECIFIC COUNTRY endpoint`**
```javascript
/v2/countries/usa                 [by default show actual data]
/v2/countries/usa?yesterday       [show yesterday data]
/v2/countries/usa?yesterday=1     [show yesterday data]
/v2/countries/usa?yesterday=0     [show actual data]
/v2/countries/usa?yesterday=true  [show yesterday data]
/v2/countries/usa?yesterday=false [show actual data]
```

### CODE REVIEW

**Comment by ebwinters** [comment]
Will look at this, make comments, and perhaps add tests tomorrow. Thanks!

**Comment by puf17640** [comment]
I love it! I am currently refactoring all scraper files!

File: routes/api_worldometers.js [hunk 1]
```diff
@@ -94,4 +94,33 @@ router.get('/yesterday/:query', async (req, res) => {
...
+	let countries = [];
+	countries = JSON.parse(await redis.get(wordToBoolean(yesterday) ? keys.yesterday : keys.countries)).splice(1).map(fixApostrophe);
+	countries = splitQuery(query)
+		.map(country => countryUtils.getCountryWorldometersData(countries, country, wordToBoolean(strict)))
+		.filter(value => value);
+	if (countries.length > 0) {
+		res.send(countries.length === 1 ? countries[0] : countries);
+		return;
+	}
+	res.status(404).send({ message: 'Country not found or doesn\'t have any cases' });
```
Comment (ebwinters) [inline routes/api_worldometers.js:123]: Why not just make it `if` `else`

File: utils/string_utils.js [hunk 2]
```diff
@@ -6,9 +6,20 @@ const wordsStandardize = (word) => {
...
 
+const wordToBoolean = word => {
+	if (word === '') return true;
+	if (word) {
+		if (word.toLowerCase() === 'true' || word === '1') return true;
+		else if (word.toLowerCase() === 'false' || word === '0') return false;
+		else return Boolean(word);
+	}
+	return false;
+};
```
Comment (ebwinters) [inline utils/string_utils.js:17]: To be honest, I think we should keep query params to two options: `?yesterday` and `yesterday=true`. There isn't really a need for `0` or `1`

**Comment by EliteDaMyth** [comment]
also dont forget the documentation

File: routes/api_worldometers.js (hunk 1 above)
Comment (buster95) [reply routes/api_worldometers.js:123]: was improved
```javascript
if (countries.length > 0) {
    return res.send(countries.length === 1 ? countries[0] : countries);
}
res.status(404).send({ message: 'Country not found or doesn\'t have any cases' });
```

File: routes/api_worldometers.js [hunk 3]
```diff
@@ -99,9 +99,10 @@ router.get('/yesterday/:query', async (req, res) => {
 router.get('/v2/all', async (req, res) => res.send(await getAllData(wordToBoolean(req.query.yesterday) ? keys.yesterday : keys.countries)));
 
 router.get('/v2/countries', async (req, res) => {
+	console.log(req.query);
 	const { sort, yesterday } = req.query;
-	let countries = [];
-	countries = JSON.parse(await redis.get(wordToBoolean(yesterday) ? keys.yesterday : keys.countries)).splice(1).map(fixApostrophe);
+	console.log(wordToBoolean(sort), sort);
```
Comment (buster95) [inline routes/api_worldometers.js]: Ethan you left a console.log

**Pull request merged**